access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -338,152 +340,63 @@ public SearchResultResponse getFacetList(SearchRequest searchRequest) {
-				rightList = queryResponse.getBeans(BriefObjectMetadataBean.class);
-			} catch (SolrServerException e) {
-				LOG.error("Error retrieving Neighboring items: " + e);
-				return null;
-			}
-
-			LOG.debug("Got " + rightList.size() + " items for right list");
-
-			// Left list
-
-			// (Note that display order stuff is reversed.)
-
-			query = new StringBuilder();
-
-			query.append(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name())).append(":[* TO ")
-					.append(pivotOrder - 1).append("]");
-			query.append(accessRestrictionClause);
-			solrQuery.setQuery(query.toString());
-
-			solrQuery.setSort(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name()), SolrQuery.ORDER.desc);
-
+		
+		solrQuery.setQuery("*:*" + accessRestrictionClause);
+		
+		addSort(solrQuery, "default", true);
+		solrQuery.setRows(NEIGHBOR_SEEK_PAGE_SIZE);
+		solrQuery.setFields("id");
+		
+		long total = -1;
+		int start = 0;
+		pageLoop: do {
 			try {
+				solrQuery.setStart(start);
 				QueryResponse queryResponse = this.executeQuery(solrQuery);
-				leftList = queryResponse.getBeans(BriefObjectMetadataBean.class);
+				total = queryResponse.getResults().getNumFound();
+				for (SolrDocument doc : queryResponse.getResults()) {
+					if (metadata.getId().equals(doc.getFieldValue("id"))) {
+						break pageLoop;
+					}
+					start++;
+				}
 			} catch (SolrServerException e) {
 				LOG.error("Error retrieving Neighboring items: " + e);
 				return null;
 			}
-
-			LOG.debug("Got " + leftList.size() + " items for left list");
-
-			// Trim the lists
-
-			int halfWindow = windowSize / 2;
-
-			// If we have enough in both lists, trim both to be
-			// halfWindow long.
-
-			if (leftList.size() >= halfWindow && rightList.size() >= halfWindow) {
-
-				LOG.debug("Trimming both lists");
-
-				leftList.subList(halfWindow, leftList.size()).clear();
-				rightList.subList(halfWindow, rightList.size()).clear();
-
-				// If we don't have enough in the left list and we have extra in the right list,
-				// try to pick up the slack by trimming fewer items from the right list.
-
-			} else if (leftList.size() < halfWindow && rightList.size() > halfWindow) {
-
-				LOG.debug("Picking up slack from right list");
-
-				// How much extra do we need from the right list?
-
-				int extra = halfWindow - leftList.size();
-
-				// Only "take" the extra (ie, clear less of the right list) if we have it available.
-
-				if (halfWindow + extra < rightList.size())
-					rightList.subList(halfWindow + extra, rightList.size()).clear();
-
-			} else if (rightList.size() < halfWindow && leftList.size() > halfWindow) {
-
-				LOG.debug("Picking up slack from left list");
-
-				int extra = halfWindow - rightList.size();
-
-				if (halfWindow + extra < leftList.size())
-					leftList.subList(halfWindow + extra, leftList.size()).clear();
-
+		} while (start < total);
+		
+		// Wasn't found, no neighbors shall be forthcoming
```

Comment:
It shouldn't, but could happen in unlikely cases such as an item having been moved, deleted, hidden during the request, or some other general error state.